spec/commands/change_spec.rb

```diff
@@ -0,0 +1,158 @@
+# frozen_string_literal: true
+
+RSpec.describe "bundle change" do
+  before :each do
+    install_gemfile <<-G
+      source "file://#{gem_repo1}"
+
+      gem "rack", "~> 1.0", :group => [:dev]
+      gem "weakling", ">=  0.0.1"
+
+      group :test do
+        gem "rack-test", "= 1.0"
+        gem "rspec"
+      end
+    G
+  end
+
+  describe "when gem is not present" do
+    it "throws error" do
+      bundle "change rake --group dev1"
+
+      expect(out).to include("`rake` could not be found in the Gemfile.")
```

Comment:
please test that failure does not change the gemfile

```ruby
expect { bundle ... }.not_to change { gemfile }
```

Reply:
No operation is done on gemfile until this operation.

Reply:
right but we should test that that's the case :)